bindings/java/src/org/sleuthkit/datamodel/CaseDatabaseFactory.java

```diff
@@ -786,8 +786,12 @@ void performPreInitialization(Connection conn) throws TskCoreException {
 		void performPostTableInitialization(Connection conn) throws TskCoreException {
 			try (Statement stmt = conn.createStatement()) {
 				stmt.execute("ALTER SEQUENCE blackboard_artifacts_artifact_id_seq minvalue -9223372036854775808 restart with -9223372036854775808");
+				
+				// CT-9000: Postgres supports composite and partial indexes which results in smaller indexes and faster inserts. 
+				// So in Postgres we can have an index which indexes only tsk_files with non-null MD5 and non-zero size:
+				stmt.execute("CREATE INDEX tsk_files_datasrc_md5_size_partial_index ON tsk_files(data_source_obj_id, md5, size) WHERE md5 IS NOT NULL AND size > 0");
 			} catch (SQLException ex) {
-				throw new TskCoreException("Error altering artifact ID sequence", ex);
+				throw new TskCoreException("Error performing PostgreSQL post table initialization", ex);
 			}
```

Comment:
I'm completely unclear on why you would add `, size` to the index after `md5`. Isn't every file with the same `md5` going to be the same size? @jayaramcs ?

Reply:
Yes, but we get a whole lot of file entries where MD5s are not calculated and is therefore NULL. It is also possible to have entries in this table with MD5s but no size - such as when we ingest flat file with paths and MD5s and not the sizes.

Reply:
Ah. Got it. You might want to add a comment to that effect. Thanks.